File: iOSDevGuide/chapter08/8.1.4/practice814/practice814/ViewController.m
//
//  ViewController.m
//  practice814
//
//  Created by aa64mac on 10/14/15.
//  Copyright © 2015 cyj. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"UITextAlignment";
    self.view.backgroundColor = [UIColor blackColor];
    UILabel *label1 = [[UILabel alloc] initWithFrame:CGRectMake(0, 10, 320, 30)];
    UILabel *label2 = [[UILabel alloc] initWithFrame:CGRectMake(0, 50, 320, 30)];
    UILabel *label3 = [[UILabel alloc] initWithFrame:CGRectMake(0, 90, 320, 30)];
    label1.textAlignment = NSTextAlignmentLeft;
    label2.textAlignment = NSTextAlignmentCenter;
    label3.textAlignment = NSTextAlignmentRight;
    label1.text = @"NSTextAlignmentLeft";
    label2.text = @"NSTextAlignmentCenter";
    label3.text = @"NSTextAlignmentRight";
    label1.textColor = [UIColor whiteColor];
    label2.textColor = [UIColor whiteColor];
    label3.textColor = [UIColor whiteColor];
    [self.view addSubview:label1];
    [self.view addSubview:label2];
    [self.view addSubview:label3];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: iOSDevGuide/chapter28/28.3/practice283/practice283/ViewController.swift
//
//  ViewController.swift
//  practice283
//
//  Created by aa64mac on 12/22/15.
//  Copyright © 2015 cyj. All rights reserved.
//

import UIKit
import CoreMotion

class ViewController: UIViewController {

    @IBOutlet weak var acc_x: UILabel!
    @IBOutlet weak var acc_y: UILabel!
    @IBOutlet weak var acc_z: UILabel!
    
    @IBOutlet weak var gyro_x: UILabel!
    @IBOutlet weak var gyro_y: UILabel!
    @IBOutlet weak var gyro_z: UILabel!
    
    @IBOutlet weak var attitude_roll: UILabel!
    @IBOutlet weak var attitude_pitch: UILabel!
    @IBOutlet weak var attitude_yaw: UILabel!
    
    @IBOutlet weak var attitude_x: UILabel!
    @IBOutlet weak var attitude_y: UILabel!
    @IBOutlet weak var attitude_z: UILabel!
    @IBOutlet weak var attitude_w: UILabel!
    
    let motionManager = CMMotionManager()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        motionManager.deviceMotionUpdateInterval = 0.05
        //motionManager.startDeviceMotionUpdatesToQueue(NSOperationQueue.currentQueue()) { (deviceManager, error) -> Void in
            
        //}
        motionManager.startDeviceMotionUpdatesToQueue(NSOperationQueue.currentQueue()!, withHandler: { (deviceManager, error) -> Void in
            let accel = deviceManager?.userAcceleration
            self.acc_x.text = String(format: "%.2f", accel!.x)
            self.acc_y.text = String(format: "%.2f", accel!.y)
            self.acc_z.text = String(format: "%.2f", accel!.z)
            
            let gyro = deviceManager?.rotationRate
            self.gyro_x.text = String(format: "%.2f", gyro!.x)
            self.gyro_y.text = String(format: "%.2f", gyro!.y)
            self.gyro_z.text = String(format: "%.2f", gyro!.z)

            let attitude = deviceManager?.attitude
            self.attitude_roll.text = String(format: "%.2f", attitude!.roll)
            self.attitude_pitch.text = String(format: "%.2f", attitude!.pitch)
            self.attitude_yaw.text = String(format: "%.2f", attitude!.yaw)
            
            let quaternion = attitude?.quaternion
            self.attitude_x.text = String(format: "%.2f", quaternion!.x)
            self.attitude_y.text = String(format: "%.2f", quaternion!.y)
            self.attitude_z.text = String(format: "%.2f", quaternion!.z)
            self.attitude_w.text = String(format: "%.2f", quaternion!.w)
        })
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    override func shouldAutorotate() -> Bool {
        return false
    }
}

